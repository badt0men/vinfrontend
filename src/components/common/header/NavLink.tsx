import {
  Box,
  Flex,
  HTMLChakraProps,
  Icon,
  useColorModeValue as mode,
} from '@chakra-ui/react'
import * as React from 'react'
import Link from 'next/link'

interface DesktopNavLinkProps extends HTMLChakraProps<'a'> {
  active?: boolean
}

const DesktopNavLink = (props: DesktopNavLinkProps) => {
  const { active, href, ...rest }:any = props

  // return (
  //   <Flex as="a"
  //     href={href}
  //     aria-current={active ? 'page' : undefined}
  //     fontWeight="medium"
  //     color={mode('gray.600', 'gray.300')}
  //     {...rest}
  //     _activeLink={{
  //       color: mode('deepBlue.500', 'skyBlue'),
  //       fontWeight: 'medium',
  //     }}
    
  //   />
  // )
  return (
    <Link
      href={href}
      {...rest}
    />
  )
}

interface MobileNavLinkProps {
  icon: React.ElementType
  children: React.ReactNode
  href?: string
}

const MobileNavLink = (props: MobileNavLinkProps) => {
  const { icon, children, href } = props
  return (
    <Flex
      as="a"
      href={href}
      m="-3"
      p="3"
      align="center"
      rounded="md"
      cursor="pointer"
      _hover={{ bg: mode('gray.50', 'gray.600') }}
    >
      <Icon as={icon} color={mode('red.100', 'red.100')} fontSize="xl" />
      <Box marginStart="3" fontWeight="medium">
        {children}
      </Box>
    </Flex>
  )
}

export const NavLink = {
  Desktop: DesktopNavLink,
  Mobile: MobileNavLink,
}
