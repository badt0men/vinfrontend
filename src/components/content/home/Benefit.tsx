import React, { useEffect } from 'react';
import {
	Box,
	Flex,
	Heading,
	VStack,
	Text,
	useColorModeValue as mode,
	Button,
	Spacer,
	HStack,
	Image,
	ButtonProps,
} from '@chakra-ui/react';
import {BoxProps, HeadingProps} from '@chakra-ui/layout';
import { RiQuestionFill } from 'react-icons/ri';
import { motion,  useAnimation} from "framer-motion"
import { useInView } from "react-intersection-observer"
import {benefitTitleVariant, benefitDescriptionVariant, benefitConsultUsVariant, buttonVariant} from './_homeAnimation'


interface ImageProps {
	light: string,
	dark: string,
}
interface manImage {
	imageUrl: string,
	imageAlt: string,
}

const manImage: manImage = {
	imageUrl: '../image/assets/manHoldingCar.jpeg',
	imageAlt: 'Man Holding Car',
};
const leftImage: ImageProps = {
	light: "url('../image/assets/rightBlob.svg')",
	dark: "url('../image/assets/darkRightBlob.svg')",
};

const MotionBox = motion<BoxProps>(Box)
const MotionHeading = motion<HeadingProps>(Heading)
const MotionButton = motion<ButtonProps>(Button)


export default function Benefit() {
	const controls = useAnimation(); //let's you take controll of when animation should start and stop
	const [ref, inView] = useInView();

	useEffect(() => {
		if (inView) {
			controls.start("visible")
		}
		else{ 
			controls.start("invisible")
		}
	}, [controls, inView])
	
	return (
		<Box maxW="7xl" mx="auto">
			<Flex
				pl={[ '8', '32' ]}
				py={[ '16', '20' ]}
				direction={[ 'column', 'row' ]}
				bgImage={mode(leftImage.light, leftImage.dark)}
				bgPosition={[ 'left 180%', '101% 1000%' ]}
				bgRepeat="no-repeat"
				bgSize={[ '0', '650px' ]}
				w="400"
				h="650"
			>
				<Box>
					<VStack>
						<Box>
							<MotionHeading
								as="h1"
								fontSize={[ '24', '48' ]}
								color={mode('deepBlue.100', 'skyBlue')}
								fontWeight="medium"
								initial="hidden"
								animate={controls}
								variants={benefitTitleVariant}
								ref={ref}
							>
								Benefits of Checking<br /> Vehicle History Report{' '}
							</MotionHeading>
							<MotionBox 
								py="8"
								fontSize={[ '15', '17' ]}
								color={mode('headingColor', 'skyBlue')}
								fontWeight="regular"
								maxW="600px"
								pr={[ '8', '0' ]}
								initial="hidden"
								animate={controls}
								variants={benefitDescriptionVariant}
								ref={ref}

								>
							<Text>
								Itâ€™s very important to know a vehicle history when you are buying a vehicle on the
								secondary market. Car history check is a must especially with such a large used
								car market in Nigeria.
								The likelihood of buying a car after an accident, fire, flood or crime is very
								high.<br/> Vehicle information will help you avoid problems.
							</Text>
							</MotionBox>
							<MotionHeading as="h1" fontWeight="medium" fontSize={['15', '16']}
							initial="hidden"
							animate={controls}
							variants={benefitConsultUsVariant}
							ref={ref}
							>
								Want a free consultation from ONGAD experts?
							</MotionHeading>
							<Spacer my="8" />
							<MotionButton
								maxW={[ '100%', '40%' ]}
								bgGradient="linear(to-t, red.200, red.100)"
								rounded="md"
								color="white"
								_hover={{ bgGradient: 'linear(to-r, red.200, red.100)' }}
								fontSize="16px"
								fontWeight="medium"
								initial="hidden"
								animate={controls}
								variants={buttonVariant}
								ref={ref}
							>
								<HStack>
									<Text>Get in touch with us </Text>
									<RiQuestionFill size="20" />
								</HStack>
							</MotionButton>
						</Box>
					</VStack>
				</Box>
				
				<MotionBox position="relative" ml={[ '0', '8' ]} mt={[ '8', '0' ]} animate={{opacity: 1}} whileHover={{rotate: 10}} whileTap={{rotate: 10}}>

					<Box
						maxW="350"
						h="470"
						bgColor="red.500"
						rounded="lg"
						pos="relative"
						top={[ '2', '3' ]}
						left={[ '2', '3' ]}
					/>

					<Image
						src={manImage.imageUrl}
						imagealt={manImage.imageAlt}
						w={350}
						h={470}
						rounded={`lg`}
						pos="relative"
						top={[ '-50%', '-96%' ]}
						left="0"
						alt={manImage.imageAlt}
					/>
					
				</MotionBox>
			</Flex>
		</Box>
	);
}
