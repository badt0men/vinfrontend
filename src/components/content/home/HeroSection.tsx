import React from 'react';
import {Box, Flex, Heading, Text, VStack, Input, Button, Stack} from '@chakra-ui/react'
import {RiSearch2Line} from 'react-icons/ri'

export default function HeroSection():JSX.Element {
  return (
    <Box w="full" mx="auto" minH="550" bgImage="url('../image/slide/car.jpeg')" bgPosition="center" bgRepeat="no-repeat" bgSize="cover" bgBlendMode="multiply">
      <Box position="absolute" bgColor="heroOverlay" w="full" h="550">
          <Box maxW="7xl" mx="auto" py={{base: '16', md: '24'}} px={{ base: '8', md: '14' }}>
               <Flex justify="end">
                      <VStack>
                      <Box>
                          <Heading as="h1" fontSize={{base: '40px', md: '68px'}} textShadow="0px 4px 4px rgba(255, 255, 255, 0.33)" color="skyBlue">Shine Your Eyes</Heading>
                      </Box>
                      <Box justify="start">
                          <Heading as="h2" fontSize={{base: '20px', md: '28px'}} py="2" color="skyBlue">Check Am Before You Buy Am</Heading>
                      </Box>
                       {/* Search box */}
                      <Box w="full" py="8">
                          <Stack direction={['column', 'row']} spacing="4">
                              <Box w={['100%', '75%']}>
                                <Input fontSize="14px" fontWeight="medium" placeholder='Enter VIN Number' color="skyBlue"  focusBorderColor="skyBlue" letterSpacings="widest"/>
                              </Box>
                              <Box>
                                <Button w={['100%', '100%']} bgGradient="linear(to-t, red.200, red.100)" rounded="md" color="white" _hover={{bgGradient: "linear(to-r, red.200, red.100)"}} fontSize="13px" fontWeight="medium">
                                  <RiSearch2Line size="26" />
                                  Search</Button>
                              </Box>
                          </Stack>
                          <Text fontSize="11px" fontStyle="italic" py="2" color="skyBlue">Click here to test with a random VIN. </Text>
                          <Box>
                              <Heading as="h4" fontSize={{base: '16', md:'18'}} color="skyBlue" fontWeight="regular" py="2" pr={{base: '4', md:'0'}}>Research any vehicle by VIN Number for free. Be Smart.</Heading>
                          </Box>
                      </Box>
                      {/* Search Box End */}
                      </VStack>
               </Flex>
          </Box>
      </Box>
    </Box>
  );
}
