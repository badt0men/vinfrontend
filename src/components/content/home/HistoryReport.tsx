import { Box, Heading, SimpleGrid, useColorModeValue as mode, VStack, Text, Image, Spacer } from '@chakra-ui/react';
import React, { useLayoutEffect } from "react";
import { BoxProps } from '@chakra-ui/layout';
import { useAnimation, motion } from "framer-motion";
import { useInView } from "react-intersection-observer";
import { historyIconsVariant, historySectionVariant } from './_homeAnimation';


export const MotionBox = motion<BoxProps>(Box)


export default function HistoryReport() {
  
  
  //controlling inView animation 
  const controls = useAnimation(); //let's you take controll of when animation should start and stop
  const [ref, inView] = useInView();
  
  useLayoutEffect(() => {
    if (inView) {
      controls.start("visible")
    }
  }, [controls, inView])
  
  return (
    <Box maxW="7xl" mx="auto">
        <MotionBox ref={ref} initial="hidden" animate={controls} variants={historySectionVariant}>
        <Box justify="center">
            <Heading as="h1" fontSize={['25', '37']} pt="20" textAlign="center" color={mode('headingColor', 'skyBlue')} fontWeight="medium">
            Shorten And Amplify Your<br/>
            Car History Report
            </Heading>
            <Heading as="h2" fontSize={['14', '22']} py="8" textAlign="center" color={mode('subHeadingColor', 'skyBlue')} fontWeight="regular">
                Meet Our Intuitive Platform
            </Heading>
        </Box>
        </MotionBox>
        
        <Box py="2">
            <Box maxW="5xl" mx="auto">
              <Box p={[4, 8]} mr={["4", "0"]}>
              <SimpleGrid columns={[2, 4]} spacing="10">
                  {boxItems.map((boxItem) => 
                    <MotionBox ref={ref} custom={boxItem.id} initial="hidden" animate={controls} variants={historyIconsVariant} whileHover="hover" whileTap="tap" key={boxItem.id}>
                    <Box bg={mode('white', 'deepBlue.100')} h="180px" w="180px" shadow="lg">
                      <VStack px="4">
                        {boxItem.icon}
                        <Heading as="h1" fontSize={['12', '14']} textAlign="center" fontWeight="medium" mt="4">{boxItem.title}</Heading>
                        <Spacer />
                        <Box border='1px' w="50%" borderStyle="dashed" />
                        <Spacer />
                        <Text fontSize={['11', '11']} textAlign="center" fontWeight="regular" color={mode('subHeadingColor', '')} >{boxItem.description}</Text>
                      </VStack>
                    </Box>
                    </MotionBox>
                    )}
                </SimpleGrid>
              </Box>
                <MotionBox>
                <Box px={['8', '16']}>
                    <Heading as="p" fontSize={['17', '19']} textAlign="center" my="8" color={mode('subHeadingColor', '')} fontWeight="light">
                    As part of our giveback to the community, ONGAD provides <b>basic free VIN check report</b> to the public to determine the vehicle is a lemon, salvaged or has flooded condition. We also provide historical mileage/sales records for free! No credit card required!
                    </Heading>
                </Box>
                </MotionBox>
            </Box>
        </Box>
    </Box>
  );
}

const searchIcon = <Image src="../image/icons/search-alt.svg" boxSize="36px" objectFit="contain" alt="searchIcon" mt="6"  />
const dataIcon = <Image src="../image/icons/data.svg" boxSize="36px" objectFit="contain" alt="dataIcon" mt="6"  />
const paymentIcon = <Image src="../image/icons/payment.svg" boxSize="36px" objectFit="contain" alt="paymentIcon" mt="6"  />
const reportIcon = <Image src="../image/icons/report.svg" boxSize="36px" objectFit="contain" alt="reportIcon" mt="6"  />


type BoxItem = {
  id: number,
  title: string,
  description: string,
  icon: JSX.Element,
}[]

const boxItems: BoxItem = [
  {
    id: 1,
    title: 'Enter your VIN',
    description: 'The only thing you need to know to get started',
    icon: searchIcon,
  },
  {
    id: 2,
    title: 'Data Search',
    description: 'The only thing you need to know to get started',
    icon: dataIcon,
  },
  {
    id: 3,
    title: 'Secure Payment',
    description: 'The only thing you need to know to get started',
    icon: paymentIcon,
  },
  {
    id: 4,
    title: 'Get your Report',
    description: 'The only thing you need to know to get started',
    icon: reportIcon,
  },
]

