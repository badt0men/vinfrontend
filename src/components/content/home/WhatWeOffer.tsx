import React, { useEffect } from 'react'
import { Text, Box, Flex, Heading, List, Image, ListItem, useColorModeValue as mode, VStack, HStack, Button, Spacer } from '@chakra-ui/react'
import {BoxProps, HeadingProps} from '@chakra-ui/layout'
import { BsArrowRight } from 'react-icons/bs'
import Link from 'next/link'
import {motion, useAnimation } from 'framer-motion'
import { useInView } from 'react-intersection-observer'
import {titleVariant, descriptionVariant, listVariant, readMoreVariant} from './_homeAnimation'

const ListIcon = <Image src="../image/icons/point.svg" boxSize="14px" objectFit="contain" alt="searchIcon"  />
    
const MotionBox = motion<BoxProps>(Box)
const MotionHeading = motion<HeadingProps>(Heading)


export default function WhatWeOffer() {
    
      //controlling inView animation 
      const controls = useAnimation(); //let's you take controll of when animation should start and stop
      const [ref, inView] = useInView();
      
      useEffect(() => {
        if (inView) {
          controls.start("visible");
        }
        else
        {
          controls.start("invisible");
        }
      }, [controls, inView]);
    
    return (
        <Box maxW="7xl" mx="auto" bg={mode('skyBlue', 'deepBlue.100')} minH={["800", "0"]} pb={['12', '0']}>
            <Flex pl={["0","16"]} py={["16", "20"]}  direction={["column", "row"]} bgImage="url('../image/icons/girlie.svg')" bgPosition={["center 100%", "center right"]} bgRepeat="no-repeat" bgSize={["400px", "750px"]}>
                <Box px={[0, 20]} pb={["44", "0"]} mb={['16','0']}>
                    <VStack>
                        <Box>
                        <MotionHeading ref={ref} initial="hidden" animate={controls} variants={titleVariant}>
                        <Text fontSize={["14","18"]} fontWeight="medium" color={mode('teal.500', 'teal.400')}>What we Offer
                        </Text>
                        </MotionHeading>
                        <MotionHeading ref={ref} initial="hidden" animate={controls} variants={descriptionVariant} >
                        <Text fontSize={["18","24"]} py="8" color={mode('headingColor', 'skyBlue')} fontWeight="medium">We provide Naija with<br/> the most comprehensive<br/> and up to date report in the market
                        </Text>
                        </MotionHeading>
                        <List spacing={3} >
                                {listText.map(list => (
                                <ListItem pt="2" key={list.id}>
                                    <MotionHeading ref={ref} whileHover="hover" custom={list.id} initial="hidden" animate={controls} variants={listVariant}>
                                    <HStack>
                                    {ListIcon} 
                                    <Text fontSize="14" fontWeight="regular"  color={mode('headingColor', 'skyBlue')}>
                                    {list.text}
                                    </Text>
                                    </HStack>
                                    </MotionHeading>
                                </ListItem>
                                ))}
                            </List>
                            <Box> 
                                    <MotionHeading ref={ref} initial="hidden" animate={controls} variants={readMoreVariant}>
                                <Text fontSize="12" textAlign="left" py="8"><Link href="/">Plus lots more...</Link> </Text>
                                    </MotionHeading>
                                <Button bgGradient="linear(to-t, deepBlue.500, deepBlue.100)" mt={["2","8"]} color="white" rounded="lg" fontWeight="medium" _hover={{bgGradient:"linear(to-r, deepBlue.100, deepBlue.500)"}}>
                                    <HStack>
                                    <Text fontSize="12">Find out more</Text> 
                                    <BsArrowRight />
                                    
                                    </HStack>
                                </Button> 
                            </Box>
                            <Spacer />
                        </Box>
                        
                    </VStack>
                </Box>
                
            </Flex>
        </Box>
    )
}


const listText = [
    {
        id: 1,
        text: 'Vehicle financial liability check',
    },
    {
        id: 2,
        text: 'Odometer check',
    },
    {
        id: 3,
        text: 'Vehicle valuation & registration details',
    },
    {
        id: 4,
        text: 'National view of the vehicle history',
    },
    {
        id: 5,
        text: 'ANCAP safety & emission ratings',
    },
    {
        id: 6,
        text: 'Damages, stolen status, and write-off check',
    },
]