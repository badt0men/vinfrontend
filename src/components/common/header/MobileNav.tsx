/* eslint-disable @next/next/no-img-element */
import {
  Box,
  Button,
  Center,
  Flex,
  Portal,
  SimpleGrid,
  useBoolean,
  useFocusOnShow,
  VStack,
  useColorModeValue as mode,
} from '@chakra-ui/react'
import { HTMLMotionProps, motion, Variants } from 'framer-motion'
import * as React from 'react'
import FocusLock from 'react-focus-lock'
import {
  HiOutlineX,
  HiMenuAlt3,
} from 'react-icons/hi'
import { RiHome5Fill } from 'react-icons/ri'
import {GiMultiDirections} from 'react-icons/gi'
import {BsPatchQuestionFill} from 'react-icons/bs'
import {FaBlog} from 'react-icons/fa'
import {MdContacts} from 'react-icons/md'
import { RemoveScroll } from 'react-remove-scroll'
import { NavLink } from './NavLink'
import { AppButton } from '../../Buttons/AppButton'

const LogoLight = <img src="/image/logo/ongadLogo.svg" width="70" alt="logo" />
const LogoDark =  <img src="/image/logo/ongadLogoALt.svg" width="70" alt="logo-alt" />

const variants: Variants = {
  show: {
    display: 'revert',
    opacity: 1,
    scale: 1,
    transition: { duration: 0.2, ease: 'easeOut' },
  },
  hide: {
    opacity: 0,
    scale: 0.98,
    transition: { duration: 0.1, ease: 'easeIn' },
    transitionEnd: { display: 'none' },
  },
}

const Backdrop = ({ show }: { show?: boolean }) => (
  <Portal>
    <motion.div
      initial={false}
      animate={show ? 'show' : 'hide'}
      transition={{ duration: 0.1 }}
      variants={{
        show: { opacity: 1, display: 'revert' },
        hide: { opacity: 0, transitionEnd: { display: 'none' } },
      }}
      style={{
        width: '100%',
        height: '100vh',
        position: 'absolute',
        background: 'rgba(0,0,0,0.2)',
        inset: 0,
      }}
    />
  </Portal>
)

const Transition = (props: HTMLMotionProps<'div'> & { in?: boolean }) => {
  const { in: inProp, ...rest } = props
  return (
    <motion.div
      {...rest}
      initial={false}
      variants={variants}
      animate={inProp ? 'show' : 'hide'}
      style={{
        transformOrigin: 'top right',
        position: 'absolute',
        width: 'calc(100% - 32px)',
        top: '24px',
        left: '16px',
        margin: '0 auto',
        zIndex: 1,
      }}
    />
  )
}

export const MobileNav = () => {
  const [show, { toggle, off }] = useBoolean()
  const ref = React.useRef<HTMLDivElement>(null)
  useFocusOnShow(ref, { visible: show, shouldFocus: true })

  return (
    <>
      <Box
        as="button"
        type="button"
        p="1"
        fontSize="2xl"
        color={mode('deepBlue', 'skyBlue')}
        onClick={toggle}
        display={{ base: 'block', lg: 'none' }}
      >
        <HiMenuAlt3 />
      </Box>

      <Transition in={show}>
        <RemoveScroll enabled={show}>
          <Backdrop show={show} />
        </RemoveScroll>
        <FocusLock disabled={!show} returnFocus>
          <Box
            bg={mode('white', 'deepBlue.500')}
            shadow="lg"
            rounded="lg"
            ref={ref}
            tabIndex={0}
            outline={0}
          >
            <Box pt="5" pb="6" px="5">
              <Flex justify="space-between" align="center">
              {mode('light', 'dark') === 'light' ? LogoLight : LogoDark}
                <Box mr="-2" mt="-2">
                  <Center
                    as="button"
                    type="button"
                    onClick={off}
                    rounded="base"
                    p="1"
                    color={mode('gray.600', 'gray.400')}
                    _hover={{ bg: mode('gray.100', 'gray.600') }}
                  >
                    <Box srOnly>Close menu</Box>
                    <HiOutlineX aria-hidden fontSize="1.5rem" />
                  </Center>
                </Box>
              </Flex>
              <SimpleGrid as="nav" gap="8" mt="8" columns={{ base: 1, sm: 2 }} onClick={toggle}>
                <NavLink.Mobile href="/">Home</NavLink.Mobile>
                <NavLink.Mobile href="guide">Guide</NavLink.Mobile>
                <NavLink.Mobile href="faq">FAQ</NavLink.Mobile>
                <NavLink.Mobile href="blog">Blog</NavLink.Mobile>
                <NavLink.Mobile href="contact">Contact</NavLink.Mobile>
                <NavLink.Mobile><span className="emphasisColor">Buy Luxury Cars</span></NavLink.Mobile>
              </SimpleGrid>
              <VStack mt="8" spacing="4">
                {/* <Button w="100%" bgGradient='linear(to-t, red.200, red.100)' rounded="md" color="white" _hover={{bgGradient: 'linear(to-r, red.200, red.100)'}} _focus={{boxShadow: "none"}}>
                  Free VIN Check
                </Button> */}
                <AppButton>
                  Free VIN Check
                </AppButton>
                <Box textAlign="center" fontWeight="medium" fontSize={12}>
                  Still a guest?{' '}
                  <Box as="a" color={mode('red.100', 'red.100')}>
                    SIGN UP
                  </Box>
                </Box>
              </VStack>
            </Box>
          </Box>
        </FocusLock>
      </Transition>
    </>
  )
}
