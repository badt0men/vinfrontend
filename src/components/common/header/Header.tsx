import {
  Box,
  Text,
  Button,
  Flex,
  HStack,
  useColorModeValue as mode,
  useColorMode,
} from '@chakra-ui/react'
import * as React from 'react'
import { MobileNav } from './MobileNav'
import { NavLink } from './NavLink'
import {FaUserPlus, FaTelegramPlane} from 'react-icons/fa'
import {IoInvertMode} from 'react-icons/io5'
import Link from 'next/link'

const LogoLight = <img src="../image/logo/ongadLogo.svg" width="70" alt="logo" />
const LogoDark =  <img src="../image/logo/ongadLogoALt.svg" width="70" alt="logo-alt" />


const Header = () => {
  const { colorMode, toggleColorMode } = useColorMode()
  return (
    <Box>
      <Box maxW="7xl" mx="auto" py={{base: '4', md: '3'}} px={{ base: '6', md: '14' }} bg={mode('white', 'deepBlue.500')} h={{base: '50', md: '10'}}>
         <Flex justify="end" fontWeight="medium" fontSize="12">
         <HStack spacing="8">
                {/* <Link href="/">Sample Requests</Link> */}
                <Link href="/business">Business</Link>
                <Link href="/careers">Careers</Link>
                <Link href="/media">Media</Link>
                <Box borderWidth="1px" borderColor={mode('deepBlue.100','skyBlue')} px="4" py="0.5" rounded="sm">
                  <Link href="/">Login</Link>
                </Box>
          
          </HStack>
          <Box as="button" ml="6" fontSize="16" p="1" rounded="md" bg={mode('gray.100', 'gray.500')} onClick={toggleColorMode}>
            {/* Color switch icon */}
            <IoInvertMode color={colorMode === 'light' ? 'Dark' : 'Light'} />
          </Box>
         </Flex>
      </Box>
      <Box as="header" bg={mode('skyBlue', 'deepBlue.100')}>
        <Box maxW="7xl" mx="auto" py="4" px={{ base: '6', md: '8' }}>
          <Flex as="nav" justify="space-between">
            <HStack spacing="32">
              <Link href="/">
                {mode('light', 'dark') === 'light' ? LogoLight : LogoDark}
              </Link>
              <HStack display={{ base: 'none', lg: 'flex' }} spacing="10" fontWeight="medium" color={mode('gray.600', 'gray.300')}>
                <NavLink.Desktop active href="/">Home</NavLink.Desktop>
                <NavLink.Desktop href="/guide">Guide</NavLink.Desktop>
                <NavLink.Desktop href="/faq">FAQ</NavLink.Desktop>
                <NavLink.Desktop href="/blog">Blog</NavLink.Desktop>
                <NavLink.Desktop href="/guide">Contact</NavLink.Desktop>
              </HStack>
            </HStack>
            <Flex align="center">
              <HStack spacing="8" display={{ base: 'none', md: 'flex' }}>

                <Button variant="outline" rounded="md" w="40" fontSize="15" fontWeight="medium" borderWidth="2px" borderColor={mode("deepBlue.100", "gray.400")}>
                    <FaUserPlus color="deepBlue.100" />
                    <Text mx="2"> SIGN UP </Text>
                </Button>
                <Button bgGradient='linear(to-t, red.200, red.100)' rounded="md" color="white" _hover={{bgGradient: 'linear(to-r, red.200, red.100)'}}>
                <FaTelegramPlane />
                    <Text mx="2"> Free VIN Check</Text>
                </Button>
              </HStack>
              <Box ml="5">
                <MobileNav />
              </Box>
            </Flex>
          </Flex>
        </Box>
      </Box>
    </Box>
  )
}


export default Header;