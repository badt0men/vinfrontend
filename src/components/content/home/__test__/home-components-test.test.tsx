import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';
import { Header } from '../../../common/header/Header';
import Benefit from '../Benefit';
import HeroSection from '../HeroSection';
import HistoryReport from '../HistoryReport';
import WhatWeOffer from '../WhatWeOffer';

describe("Homepage Test", () => {
  const header = shallow(<Header />);
  const hero = shallow(<HeroSection />);
  const history = shallow(<HistoryReport />);
  const what_we_offer = shallow(<WhatWeOffer />);
  const benefits = shallow(<Benefit />);
  test("should render all 5 componenets correctly", () => {
      expect(header).toMatchSnapshot()
      expect(hero).toMatchSnapshot()
      expect(history).toMatchSnapshot()
      expect(what_we_offer).toMatchSnapshot()
      expect(benefits).toMatchSnapshot() 
  
  })
})

