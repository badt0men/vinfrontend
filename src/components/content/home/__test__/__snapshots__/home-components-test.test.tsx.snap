// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Test should render all 5 \`{five}\` componenets correctly 1`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 \`{five}\` componenets correctly 2`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 \`{five}\` componenets correctly 3`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 \`{five}\` componenets correctly 4`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 \`{five}\` componenets correctly 5`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 componenets correctly 1`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 componenets correctly 2`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 componenets correctly 3`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 componenets correctly 4`] = `ShallowWrapper {}`;

exports[`Homepage Test should render all 5 componenets correctly 5`] = `ShallowWrapper {}`;
